core/src/main/java/com/yahoo/imapnio/async/data/FetchResult.java

```diff
@@ -0,0 +1,57 @@
+package com.yahoo.imapnio.async.data;
+
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.sun.mail.imap.protocol.FetchResponse;
+
+/**
+ * This class provides the highest modification sequence and the list of fetch response from fetch command response.
+ */
+public class FetchResult {
+
+    /** The highest modification sequence, only shown when CondStore is enabled. */
+    private Long highestModSeq;
+
+    /** The collection of fetch response. */
+    private final List<FetchResponse> fetchResponses;
```

Comment:
As mentioned in my earlier comment, not all responses will be a FetchResponse (some could be EXPUNGE or VANISHED response), so this should be IMAPResponse